src/models/layerChanges.ts

```diff
@@ -0,0 +1,239 @@
+export interface Link {
+    id: number;
+    questionRequired: boolean;
+    questionVisibility: QuestionVisibilityStatus;
+    entrypointTitle: string;
+    segmentTitle: string;
+    questionTitle: string;
+    conflict: boolean;
+}
+interface QuestionActivatorLinkMap {
+    [key: string]: Link[] | undefined;
+}
+
+interface LayerChanges {
+    id: number;
+    parentLayerId: number;
+    lastCompareLayerId: number | null;
+    conflict: boolean;
+    entrypointMap: ChangesMap<EntryPointList>;
+    segmentMap: ChangesMap<SegmentsList>;
+    questionMap: ChangesMap<QuestionList>;
+    questionActivatorLinkMap: QuestionActivatorLinkMap;
+    usedEntrypointTitleList: string[];
+}
+
+interface LayerChangesState {
+    item: LayerChanges | null;
+    isLoading: boolean;
+    error: ApiError | null;
+}
+
+const fetchLayerChanges = createAsyncThunk<LayerChanges, number, { rejectValue: ApiError }>(
+    'layerChanges/fetchLayerChanges',
+    async (id, thunkApi) => {
+        let response: AxiosResponse<LayerChanges>;
+
+        try {
+            response = await api.get<LayerChanges>(`${GET_LAYERS_URL}/${id}/${LAYER_CHANGES_PATH}`);
+        } catch (error) {
+            return thunkApi.rejectWithValue(apiErrorHandler(error));
+        }
+
+        return response.data;
+    }
+);
+
+const initialState: LayerChangesState = {
+    item: null,
+    isLoading: true,
+    error: null,
+};
+
+const layerChangesSlice = createSlice({
+    name: 'layerChanges',
+    initialState,
+    reducers: {
+        reset: (state) => {
+            state = initialState;
+            return state;
+        },
+    },
+    extraReducers: (builder) => {
+        builder
+            .addCase(fetchLayerChanges.pending, (state) => {
+                state.isLoading = true;
+                state.error = null;
+            })
+            .addCase(fetchLayerChanges.fulfilled, (state, action) => {
+                state.isLoading = false;
+                state.item = action.payload;
+            })
+            .addCase(fetchLayerChanges.rejected, (state, action) => {
+                state.isLoading = false;
+
+                if (action.payload !== undefined) {
+                    state.error = action.payload;
+                } else {
+                    state.error = { message: 'Произошла непредвиденная ошибка' };
+                }
+            });
+    },
+});
+
+const selectLayerChangesError = (state: RootState): ApiError | null => state.currentLayerChanges.error;
+const selectLayerChangesLoadingStatus = (state: RootState): boolean => state.currentLayerChanges.isLoading;
+const selectCreatedSegments = (state: RootState): SegmentsList | null => {
+    if (state.currentLayerChanges.item === null) {
+        return null;
+    }
+    if (state.currentLayerChanges.item.segmentMap.CREATED === undefined) {
+        return null;
+    }
+    if (state.currentLayerChanges.item.segmentMap.CREATED.length === 0) {
+        return null;
+    }
```

Comment:
давай схлопнем эти условия
```suggestion
    if (!state.currentLayerChanges.item?.segmentMap.CREATED?.length) {
        return null;
    }
```

Reply:
Исправил.